### PR TITLE
practice4-task3_MessageBus

### PR DESCRIPTION
**Pull request by tyroff**
Task 3. (20 points) «Where’s your bus, dude? »

Implement message bus using Producer-Consumer pattern.
1. Implement asynchronous message bus. Do not use queue implementations from java.util.concurrent.
2. Implement producer, which will generate and post randomly messages to the queue
3. Implement consumer, which will consume messages on specific topic and log to the console message payload

Optional: 4. Application should create several consumers and producers that run in parallel.

### CODE REVIEW

File: src/practice4/task3_MessageBus/util/MyBlockingQueue.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * @param <E> the type of object added to the LinkedList
+ * @author Kiryl_Vinakurau
+ */
+public final class MyBlockingQueue<E> implements MyQueue<E>, java.io.Serializable {
+
+    private List queue = new LinkedList();
+    private int limit = 1;
+
+    public MyBlockingQueue(int limit) {
+        this.limit = --limit;
```
Comment (UShalamitski) [inline src/practice4/task3_MessageBus/util/MyBlockingQueue.java:20]: Why "--limit" ?

File: src/practice4/task3_MessageBus/entity/BusMessage.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * This class contains information about the bus: bus number, date and time of departure.
+ *
+ * @author Kiryl_Vinakurau
+ */
+public class BusMessage {
```
Comment (UShalamitski) [inline src/practice4/task3_MessageBus/entity/BusMessage.java:12]: "Message bus" is just a pattern name

File: src/practice4/task3_MessageBus/util/MyBlockingQueue.java (hunk 1 above)
Comment (tyroff) [reply src/practice4/task3_MessageBus/util/MyBlockingQueue.java:20]: I experimented, forgot to clean.

File: src/practice4/task3_MessageBus/entity/BusMessage.java (hunk 2 above)
Comment (tyroff) [reply src/practice4/task3_MessageBus/entity/BusMessage.java]: Name of class BusMesage changed to TransportTimetable

**Pull request merged**